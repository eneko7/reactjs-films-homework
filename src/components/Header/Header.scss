.app_header {
  width: 100%;
  height: 88vh;
  padding-top: 20px;
  background-image: url(../../images/cover-image.jpg);
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  position: relative;
  z-index: 3;
}

.app_header.error {
  height: 7vh;
  background: none !important;
}

.app_header::after {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.4);
  z-index: 2;
}

@media (max-width: 767px) {
  .app_header {
    height: 100vh;
  }
}

@media (max-width: 621px) {
  .app_header.error {
    height: 15vh;
    background: none !important;
  }
}
