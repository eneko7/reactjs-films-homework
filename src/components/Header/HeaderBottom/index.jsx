export {dafault} from './HeaderBottom.jsx';