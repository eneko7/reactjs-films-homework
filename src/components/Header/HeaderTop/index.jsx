import HeaderTop from './HeaderTopContainer';

export default HeaderTop;
