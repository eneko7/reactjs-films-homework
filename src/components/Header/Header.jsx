import React from 'react';
import PropTypes from 'prop-types';
import style from './Header.scss';
import HeaderBottom from './HeaderBottom/HeaderBottom';
import HeaderTopContainer from './HeaderTop/HeaderTopContainer';

const Header = (props) => {
  const { films, genres } = props;
  const film = films[0];
  let bg = {};
  let filmGenres = [];
  let filmName = '';
  let filmDescription = '';
  let filmRate = 0;
  let filmId = 0;
  if (typeof film === 'object') {
    bg = { backgroundImage: `url(http://image.tmdb.org/t/p/w1280${film.backdrop_path})` };
    filmGenres = film.genre_ids;
    filmName = film.original_title;
    filmDescription = film.overview;
    filmRate = film.vote_average;
    filmId = film.id;
  }
  return (
    <header id="app_header" className={style.app_header} style={bg}>
      <HeaderTopContainer />
      <HeaderBottom
        filmGenres={filmGenres}
        filmName={filmName}
        filmDescription={filmDescription}
        filmRate={filmRate}
        allGenres={genres}
        filmId={filmId}
      />
    </header>
  );
};

Header.propTypes = {
  films: PropTypes.arrayOf(PropTypes.object).isRequired,
  genres: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Header;
