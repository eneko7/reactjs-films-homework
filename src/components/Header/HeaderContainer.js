import { connect } from 'react-redux';
import Header from './Header';
import { getFilms } from '../../modules/films/filmsSelectors';
import { getGenres } from '../../modules/genres/genresSelectors';

const mapStateToProps = state => ({
  films: getFilms(state),
  genres: getGenres(state),
});

const mapDispatchToProps = {
};
export default connect(mapStateToProps, mapDispatchToProps)(Header);
