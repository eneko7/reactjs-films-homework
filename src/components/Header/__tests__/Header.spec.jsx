import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Header from '../Header';

const shallow = new ShallowRenderer();

const data = {
  films: [
    {
      allGenres: [{ id: 28, name: 'Action' }, { id: 12, name: 'Adventure' }, { id: 16, name: 'Animation' }],
      filmId: 23355,
      filmName: 'name',
      filmGenres: [1, 2, 3],
      filmRate: 5,
      filmDescription: 'name',
    },
  ],
  genres: [
    { id: 28, name: 'Action' }, { id: 12, name: 'Adventure' }, { id: 16, name: 'Animation' },
  ],
};

const badData = {
  films: '',
  genres: [
    { id: 28, name: 'Action' }, { id: 12, name: 'Adventure' }, { id: 16, name: 'Animation' },
  ],
};

describe('Header', () => {
  it('render correctly ', () => {
    shallow.render(<Header {...data} />);
    const tree = shallow.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it('render correctly if film empty ', () => {
    shallow.render(<Header {...data} />);
    const tree = shallow.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
  it('render correctly ', () => {
    shallow.render(<Header {...badData} />);
    const tree = shallow.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it('render correctly if film empty ', () => {
    shallow.render(<Header {...badData} />);
    const tree = shallow.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
