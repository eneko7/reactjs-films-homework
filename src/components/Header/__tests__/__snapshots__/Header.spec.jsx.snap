// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render correctly  1`] = `
<header
  className="app_header"
  id="app_header"
  style={
    Object {
      "backgroundImage": "url(//image.tmdb.org/t/p/w1280undefined)",
    }
  }
>
  <Connect(HeaderTop) />
  <Unknown
    bg={
      Object {
        "backgroundImage": "url(//image.tmdb.org/t/p/w1280undefined)",
      }
    }
    films={
      Array [
        Object {
          "allGenres": Array [
            Object {
              "id": 28,
              "name": "Action",
            },
            Object {
              "id": 12,
              "name": "Adventure",
            },
            Object {
              "id": 16,
              "name": "Animation",
            },
          ],
          "filmDescription": "name",
          "filmGenres": Array [
            1,
            2,
            3,
          ],
          "filmId": 23355,
          "filmName": "name",
          "filmRate": 5,
        },
      ]
    }
    genres={
      Array [
        Object {
          "id": 28,
          "name": "Action",
        },
        Object {
          "id": 12,
          "name": "Adventure",
        },
        Object {
          "id": 16,
          "name": "Animation",
        },
      ]
    }
  />
</header>
`;

exports[`Header render correctly  2`] = `
<header
  className="app_header error"
  id="app_header"
>
  <Connect(HeaderTop) />
</header>
`;

exports[`Header render correctly if film empty  1`] = `
<header
  className="app_header"
  id="app_header"
  style={
    Object {
      "backgroundImage": "url(//image.tmdb.org/t/p/w1280undefined)",
    }
  }
>
  <Connect(HeaderTop) />
  <Unknown
    bg={
      Object {
        "backgroundImage": "url(//image.tmdb.org/t/p/w1280undefined)",
      }
    }
    films={
      Array [
        Object {
          "allGenres": Array [
            Object {
              "id": 28,
              "name": "Action",
            },
            Object {
              "id": 12,
              "name": "Adventure",
            },
            Object {
              "id": 16,
              "name": "Animation",
            },
          ],
          "filmDescription": "name",
          "filmGenres": Array [
            1,
            2,
            3,
          ],
          "filmId": 23355,
          "filmName": "name",
          "filmRate": 5,
        },
      ]
    }
    genres={
      Array [
        Object {
          "id": 28,
          "name": "Action",
        },
        Object {
          "id": 12,
          "name": "Adventure",
        },
        Object {
          "id": 16,
          "name": "Animation",
        },
      ]
    }
  />
</header>
`;

exports[`Header render correctly if film empty  2`] = `
<header
  className="app_header error"
  id="app_header"
>
  <Connect(HeaderTop) />
</header>
`;
